### PR TITLE
Implement mocking response in test mode

### PR DESCRIPTION
**Pull request by mhabedini**

### CODE REVIEW

**Comment by Hebrahimzadeh** [comment]
Please write some tests for this feature.

File: src/Requests/PendingRequest.php [hunk 1]
```diff
@@ -145,8 +129,16 @@ private function fullUrl($path): string
         return $baseUrl.($servicePath === '' ? $servicePath : '/'.$servicePath).'/'.$finalPath;
     }
 
-    private function respond($result)
+    private function respond($url, $data, $method)
     {
+        if ($this->factory instanceof Http && $this->factory->getMockPath() && app()->runningUnitTests()) {
```
Comment (omalizadeh) [inline src/Requests/PendingRequest.php:134]: better to check if it's testing mode first. if not, other conditions are not checked

File: src/Responses/Mock.php [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+{
+    public static function fakeResponse($jsonPath): Response
+    {
+        $mockDirectory = base_path().'\tests\mock\\';
+        $jsonFile = file_get_contents($mockDirectory.$jsonPath);
+        $json = json_decode($jsonFile, true, 512, JSON_THROW_ON_ERROR);
+        Http::fake([
+            'test' => $json
+        ]);
+        return Http::get('test');
```
Comment (omalizadeh) [inline src/Responses/Mock.php:18]: i don't understand this piece of code and 'test' string.
please update README so we can check what's this all about and how it is going to work

**Review by omalizadeh — CHANGES_REQUESTED** [review]
Update readme

**Pull request merged**